serverAddons/PRA3/addons/PRA3_Server/Revive/HitEffects/fn_unconscious.sqf

```diff
@@ -0,0 +1,165 @@
+#include "macros.hpp"
+/*
+    Project Reality ArmA 3
+
+    Author: BadGuy, joko // Jonas, NetFusion
+
+    Description:
+    Handles unconsciousness.
+
+    Parameter(s):
+    PFH Return
+
+    Returns:
+    None
+*/
+["missionStarted", {
+    ["Killed", {
+        (_this select 0) params ["_unit", "_killer"];
+
+        setPlayerRespawnTime 10e10;
+
+        private _preventInstantDeath = [QGVAR(Settings_preventInstantDeath), 1] call CFUNC(getSetting);
+
+        // Abort if the respawn button was pressed or unconsciousness
+        if (_preventInstantDeath == 0 || (time - (missionNamespace getVariable ["RscDisplayMPInterrupt_respawnTime",-1])) < 1) exitWith {
+            [QGVAR(Killed), [PRA3_Player, _killer]] call CFUNC(localEvent);
+        };
+
+        if (!(PRA3_Player getVariable [QGVAR(isUnconscious), false])) then {
+            private _gear = [PRA3_Player] call CFUNC(saveGear);
+            PRA3_Player setVariable [QGVAR(killer), _killer];
+
+            [{
+                private _corpse = PRA3_Player;
+
+                //@todo this may trigger an unwanted respawn event - other modules may reset their variables on respawn
+                [playerSide, group PRA3_Player, getPosWorld PRA3_Player] call CFUNC(respawn);
+
+                [_this, PRA3_Player] call CFUNC(restoreGear);
+                PRA3_Player setDir getDir _corpse;
+                deleteVehicle _corpse;
+
+                ["UnconsciousnessChanged", true] call CFUNC(localEvent);
+            }, 3, _gear] call CFUNC(wait);
+        } else {
+            if (isNull _killer) then {
+                _killer = PRA3_Player getVariable [QGVAR(killer), objNull];
+            };
+            [QGVAR(Killed), [PRA3_Player, _killer]] call CFUNC(localEvent);
+        };
+    }] call CFUNC(addEventHandler);
+}] call CFUNC(addEventHandler);
+
+// Reset values on death
+[QGVAR(Killed), {
+    (_this select 0) params ["_unit"];
+
+    ["UnconsciousnessChanged", false] call CFUNC(localEvent);
+}] call CFUNC(addEventHandler);
+
+["UnconsciousnessChanged", {
+    (_this select 0) params ["_state"];
+
+    PRA3_Player setVariable [QGVAR(isUnconscious), _state, true];
+    [_state] call CFUNC(disableUserInput);
+
+    if (_state) then {
+        if ((vehicle PRA3_Player) isKindOf "StaticWeapon") then {
```

Comment:
why only checking StaticWeapon? we can not pull players out of Vehicles currently